Assets/SEE/XR/RadialSelection.cs

```diff
@@ -0,0 +1,409 @@
+    /// </summary>
+    public void HideAndTriggerSelected()
+    {
+        radialPartCanvas.gameObject.SetActive(false);
+        OnPartSelected.Invoke(currentSelectedRadialPart);
+        IndicatorChange = true;
+    }
+    /// <summary>
+    /// This method calculates, at which radial-part the user is aiming at.
+    /// </summary>
+    public void GetSelectedRadialPart()
+    {
+        Vector3 centerToHand = handTransform.position - radialPartCanvas.position;
+        Vector3 centerToHandProjected = Vector3.ProjectOnPlane(centerToHand, radialPartCanvas.forward);
+
+        float angle = Vector3.SignedAngle(radialPartCanvas.up, centerToHandProjected, -radialPartCanvas.forward);
+
+        if (angle < 0)
+        {
+            angle += 360;
+        }
+
+        currentSelectedRadialPart = (int)angle * numberOfRadialPart / 360;
+
+        for (int i = 0; i < spawnedParts.Count; i++)
+        {
+            if (i == currentSelectedRadialPart)
+            {
+                spawnedParts[i].GetComponent<Image>().color = Color.yellow;
+                spawnedParts[i].transform.localScale = 1.1f * Vector3.one;
+                spawnedParts[i].gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().color = Color.yellow;
+                spawnedParts[i].gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().fontStyle = (FontStyles)FontStyle.Bold;
+            }
+            else
+            {
+                spawnedParts[i].GetComponent<Image>().color = Color.white;
+                spawnedParts[i].transform.localScale = Vector3.one;
+                spawnedParts[i].gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().color = Color.black;
+                spawnedParts[i].gameObject.transform.Find("TextField").gameObject.MustGetComponent<TextMeshProUGUI>().fontStyle = (FontStyles)FontStyle.Normal;
+            }
+        }
+    }
+    /// <summary>
+    /// Is true if the radial menu is open or not.
+    /// </summary>
+    bool spawned;
+    /// <summary>
+    /// This method activates the selected action, or opens a submenu/mainmenu.
+    /// </summary>
+    /// <param name="i"></param>
+    public void SelectAction(int i)
+    {
+        if (menuEntrys.Count != 0)
+        {
+            if (actions[i] == "Back")
+            {
+                actions.Clear();
+                actions.AddRange(menuEntrys);
+                numberOfRadialPart = actions.Count();
+                radialMenuTrigger = true;
+                menuEntrys.Clear();
+                subMenuPosition = radialPartCanvas.position;
+                subMenuRotation = radialPartCanvas.rotation;
+            }
+            if (actions[i] == "HideAll")
+            {
+                HideMode = HideModeSelector.HideAll;
+                GlobalActionHistory.Execute((ActionStateType)ActionStateTypes.AllRootTypes.AllElements().FirstOrDefault(a => a.Name == "Hide"));
```

Comment:
The second line is always the same in these if statements, except for the final else. It may not be easily possible to pull it out of the if blocks here, but can we at least put this statement into a local method so we only have to call it in each block instead of writing out the full thing each time?